numpy/ma/core.py

```diff
@@ -3492,7 +3492,8 @@ def dtype(self):
 
     @dtype.setter
     def dtype(self, dtype):
-        super(MaskedArray, type(self)).dtype.__set__(self, dtype)
```

Comment:
I'm confused - isn't it OK to have deprecation warnings also for `MaskedArray`?

Reply:
There is a valid code path from numpy where the dtype of a masked array is updated (I think it was the `PyArray_View`) and the refcount is already larger than 1. In that case we do not want to generate a warning, hence the use of `_set_dtype`. 
The downside of this approach is that if a user changes the dtype, we will not get a deprecation warning.

Reply:
This is why I was suggesting to have the warning only for `ndarray` proper -- then we can just not worry about `MaskedArray` and `recarray` yet.